### PR TITLE
AWS: Add TagSession support in AssumeRoleAwsClientFactory

### PR DESCRIPTION
**Pull request by xiaoxuandev**

### CODE REVIEW

File: aws/src/integration/java/org/apache/iceberg/aws/TestAssumeRoleAwsClientFactory.java [hunk 1]
```diff
@@ -68,13 +68,16 @@ public void before() {
...
-            "\"Action\": \"sts:AssumeRole\"}]}")
+            "\"Action\": [\"sts:AssumeRole\"," +
+            "\"sts:TagSession\"]}]}")
         .maxSessionDuration(3600)
         .build());
     assumeRoleProperties = Maps.newHashMap();
     assumeRoleProperties.put(AwsProperties.CLIENT_FACTORY, AssumeRoleAwsClientFactory.class.getName());
     assumeRoleProperties.put(AwsProperties.CLIENT_ASSUME_ROLE_REGION, "us-east-1");
     assumeRoleProperties.put(AwsProperties.CLIENT_ASSUME_ROLE_ARN, response.role().arn());
+    assumeRoleProperties.put(AwsProperties.CLIENT_ASSUME_ROLE_TAGS_PREFIX + "key1", "value1");
```
Comment (jackye1995) [inline aws/src/integration/java/org/apache/iceberg/aws/TestAssumeRoleAwsClientFactory.java:79]: can we test with multiple tags?

Comment (xiaoxuandev) [reply aws/src/integration/java/org/apache/iceberg/aws/TestAssumeRoleAwsClientFactory.java:79]: added. thanks.

**Review by jackye1995 — APPROVED** [review]
looks good to me, waiting for CI pass

**Comment by jackye1995** [comment]
cc @arminnajafi @rajarshisarkar @singhpk234 @amogh-jahagirdar @xiaoxuandev @yyanyy

File: aws/src/main/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactory.java [hunk 2]
```diff
@@ -104,4 +110,11 @@ public void initialize(Map<String, String> properties) {
   private String genSessionName() {
     return String.format("iceberg-aws-%s", UUID.randomUUID());
   }
+
+  private static Set<Tag> toTags(Map<String, String> properties) {
```
Comment (rajarshisarkar) [inline aws/src/main/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactory.java:114]: We have a similar method in `AwsProperties`. Can we re-use that?

https://github.com/apache/iceberg/blob/d2bb30f863d7276097c39754b9501ba27403b73c/aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java#L459-L464

Comment (jackye1995) [reply aws/src/main/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactory.java:114]: I think we cannot reuse, this is STS tag, not S3 tag, different class.

**Review by singhpk234 — APPROVED** [review]
LGTM, Thanks @xiaoxuandev for this !!!

**Pull request merged**